docs/DeveloperGuide.md

```diff
@@ -1,38 +1,337 @@
 
-* *glossary item* - Definition
+<br><br><br>
 
-## Instructions for manual testing
+## 5. Design
+
+The following architecture diagram provides a high level overview of the main components of Mod Happy and how they interact with one another.
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Components.puml)
+
+The `Main` class is responsible for handling program initialisation, termination, as well as the application's main execution logic.
+
+Mod Happy's components are summarised below:
+
+* `UI`: Manages the application's text UI.
+* `Parser`: Interprets user input and returns corresponding `Command` objects.
+* `Command`: Handles command execution logic.
+* `Data`: Manages module and task data in program memory.
+* `Storage`: Reads data from, and writes data to Mod Happy's data storage files.
+
+<br>
+
+### 5.1. UI Component
+
+The `TextUi` class serves strictly as intermediary between the user and the program, and does not directly interact with any components other than the `Main` class. It fulfils the following roles:
+
+- Listens for and grabs the user's input using a `Scanner`, and returns it to `Main` as a string for further processing.
+- Displays any command results or status and error messages to the user.
+
+<br>
+
+### 5.2. Parser Component
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Parser.puml)
+
+The `Parser` component serves to interpret user input and construct the relevant `Command` objects to be returned to `Main` for execution later on. This component comprises the following classes:
+
+* `Parser`, which is an abstract class serving as the parent of all other classes in this component. This class implements basic parsing functionality through the use of regular expressions and the built-in `Pattern` class. This functionality is modified by its child classes to suit their own purposes.
+* `ModHappyParser`, which identifies the command word present in the user input and invokes the relevant command-specific parser.
+* A variety of command-specific parsers (e.g. `AddParser` for the `add` command), referred to in this guide as `XYZParser` for simplicity. These classes perform further parsing on any command-specific arguments, constructing and returning the corresponding `XYZCommand` object.
+
+> 📔 <span style="color:#00bb00">**NOTE:**</span>
+> 
+> `NoArgumentParser` is an exception to the above; instead of being associated with a single command type, it is responsible for handling all commands which do not accept any arguments.
+
+The following details how the `Parser` component works at runtime:
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ParserSequenceDiagram.puml)
```

Comment:
Since you have mentioned the exception to be thrown above, is it possible for the same exception to be thrown here as well? (i.e. targetModule of wrong format, missing fields). 
![image](https://user-images.githubusercontent.com/69446722/161180885-d65cbb25-87cc-4012-b8cd-4063e244d12d.png)